lib/render/SliceRenderer.cpp

```diff
@@ -435,34 +524,71 @@ void SliceRenderer::_getModifiedExtents(vector<double> &min, vector<double> &max
     box->GetExtents(min, max);
     VAssert(min.size() == 3);
     VAssert(max.size() == 3);
-
-    vector<double> sampleLocation = p->GetValueDoubleVec(p->SampleLocationTag);
-    VAssert(sampleLocation.size() == 3);
-    switch (box->GetOrientation()) {
-    case XY:
-        min[Z] = sampleLocation[Z];
-        max[Z] = sampleLocation[Z];
-        break;
-    case XZ:
-        min[Y] = sampleLocation[Y];
-        max[Y] = sampleLocation[Y];
-        break;
-    case YZ:
-        min[X] = sampleLocation[X];
-        max[X] = sampleLocation[X];
-        break;
-    default: VAssert(0);
-    }
 }
 
 int SliceRenderer::_paintGL(bool fast)
 {
     int rc = 0;
 
+    glDepthMask(GL_TRUE);
+    glEnable(GL_DEPTH_TEST);
+
+
+#ifdef DEBUG
+    // 3D green polygon that shows where we should see data
+    LegacyGL *lgl = _glManager->legacy;
+    lgl->Color4f(0, 1., 0, 1.);
+    lgl->Begin(GL_LINES);
+    if (_polygon3D.size()) {
+        for (int i = 0; i < _polygon3D.size() - 1; i++) {
+            glm::vec3 vert1 = _polygon3D[i];
+            glm::vec3 vert2 = _polygon3D[i + 1];
+            lgl->Vertex3f(vert1.x, vert1.y, vert1.z);
+            lgl->Vertex3f(vert2.x, vert2.y, vert2.z);
+        }
+        lgl->Color4f(0, 1., 0, 1.);
+        glm::vec3 vert1 = _polygon3D[_polygon3D.size() - 1];
+        glm::vec3 vert2 = _polygon3D[0];
+        lgl->Vertex3f(vert1.x, vert1.y, vert1.z);
+        lgl->Vertex3f(vert2.x, vert2.y, vert2.z);
+    }
+    lgl->End();
+
+    // 3D yellow enclosing rectangle that defines the perimeter of our texture
+    // This can and often will extend beyond the Box
+    lgl = _glManager->legacy;
+    lgl->Begin(GL_LINES);
+    lgl->Color4f(1., 1., 0., 1.);
+    if (_rectangle3D.size()) {
+        for (int i = 0; i < _rectangle3D.size() - 1; i++) {
+            glm::vec3 vert1 = _rectangle3D[i];
+            glm::vec3 vert2 = _rectangle3D[i + 1];
+            lgl->Vertex3f(vert1.x, vert1.y, vert1.z);
+            lgl->Vertex3f(vert2.x, vert2.y, vert2.z);
+        }
+        glm::vec3 vert1 = _rectangle3D[3];
+        glm::vec3 vert2 = _rectangle3D[0];
+        lgl->Vertex3f(vert1.x, vert1.y, vert1.z);
+        lgl->Vertex3f(vert2.x, vert2.y, vert2.z);
+    }
+    lgl->End();
+#endif
```

Comment:
Could this be moved to a function for readability?

Reply:
Sure, no problem.

Reply:
@stasj I'm not sure why, but this comment is not getting an "Outdated" label for the code it's referring to.  You can see the change [here](https://github.com/NCAR/VAPOR/blob/a8c58b18bbb21e5036531de1c8cabc80d6d44574/lib/render/SliceRenderer.cpp#L535).